app/models/order_item.rb

```diff
@@ -11,19 +11,16 @@ class OrderItem < ActiveRecord::Base
   validates :product_id, presence: true, numericality: { only_integer: true, greater_than: 0 }
   validates :order_id, presence: true, numericality: { only_integer: true, greater_than: 0 }
   validates :quantity_ordered, presence: true, numericality: { only_integer: true, greater_than: 0 }
-  validate :order_item_is_unique? # OPTIMIZE: I think this makes the above before_create obsolete?!
+  validate :order_item_is_unique?
 
 
+  # mutative methods
+
   def more!
-    reload # removing this line == DANGER WILL ROBINSON
```

Comment:
we aren't doing a complicated check for pending orders. something about that calculation required the reload.